Mathlib/LinearAlgebra/TensorProduct/Subalgebra.lean

```diff
@@ -128,22 +126,104 @@ theorem comm_trans_rTensorBot :
     (Algebra.TensorProduct.comm R _ _).trans A.rTensorBot = A.lTensorBot :=
   AlgEquiv.toLinearEquiv_injective (toSubmodule A).comm_trans_rTensorOne
 
+end Subalgebra
+
+namespace Algebra.TensorProduct
+
+variable (R S T)
+
+/-- Given `R`-algebras `S,T`, there is a natural `R`-linear isomorphism from `S ⊗[R] T` to
+`S' ⊗[R] T'` where `S',T'` are the images of `S,T` in `S ⊗[R] T` respectively.
+This is promoted to an `R`-algebra isomorphism `Algebra.TensorProduct.algEquivIncludeRange`. -/
+def linearEquivIncludeRange :
+    S ⊗[R] T ≃ₗ[R] (includeLeft : S →ₐ[R] S ⊗[R] T).range ⊗[R]
+      (includeRight : T →ₐ[R] S ⊗[R] T).range := .ofLinear
+  (_root_.TensorProduct.map
+    includeLeft.toLinearMap.rangeRestrict includeRight.toLinearMap.rangeRestrict)
+  ((LinearMap.range includeLeft).mulMap (LinearMap.range includeRight))
+  (_root_.TensorProduct.ext' <| by
+    rintro ⟨x', x, rfl : x ⊗ₜ 1 = x'⟩ ⟨y', y, rfl : 1 ⊗ₜ y = y'⟩
+    rw [LinearMap.comp_apply, LinearMap.id_apply]
+    erw [Submodule.mulMap_tmul]
+    rw [tmul_mul_tmul, mul_one, one_mul, _root_.TensorProduct.map_tmul]
+    rfl)
+  (_root_.TensorProduct.ext' fun x y ↦ by
+    rw [LinearMap.comp_apply, LinearMap.id_apply, _root_.TensorProduct.map_tmul]
+    erw [Submodule.mulMap_tmul]
```

Comment:
Do you have some ideas on these two `erw`s? I tried `rw` and `simp` but none of them works.

Reply:
Maybe we should add `Subalgebra.mulMap_tmul`?

Reply:
In this construction no `Subalgebra.mulMap` is involved (yet). So I have absolutely no idea why it fails.

Reply:
![image](https://github.com/user-attachments/assets/e5742ab5-d0d2-4f8a-be09-cf3989a0feb7)
This shows `⟨x ⊗ₜ[R] 1, ⋯⟩` is in a subalgebra, not a submodule.

Reply:
If you write `includeLeft.toSubmodule.range` then you can probably use `Submodule.mulMap_tmul` without `erw`, but you may not want to.

Reply:
I tried `rintro (⟨x', x, rfl : x ⊗ₜ 1 = x'⟩ : ↥(LinearMap.range includeLeft) (⟨y', y, rfl : 1 ⊗ₜ y = y'⟩ : ↥(LinearMap.range includeRight))` but `rw` is still not working. I think maybe we should ignore this issue.

Reply:
I think the erw is unavoidable given that the defeq between AlgHom.range and LinearMap.range is exploited, unless lemmas specific to this case are introduced. I would merge this as is. Thanks!
bors r+